pkg/loop/plugin_median.go

```diff
@@ -0,0 +1,484 @@
+package loop
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"github.com/smartcontractkit/libocr/offchainreporting2/reportingplugin/median"
+	libocr "github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"google.golang.org/grpc"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	pb "github.com/smartcontractkit/chainlink-relay/pkg/loop/internal/pb"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+// PluginMedianName is the name for [PluginMedian]/[NewGRPCPluginMedian].
+const PluginMedianName = "median"
+
+type PluginMedian interface {
+	NewMedianPluginFactory(ctx context.Context, provider types.MedianProvider, dataSource, juelsPerFeeCoin median.DataSource, errorLog ErrorLog) (libocr.ReportingPluginFactory, error)
+}
+
+type ErrorLog interface {
+	SaveError(ctx context.Context, msg string) error
+}
+
+func PluginMedianHandshakeConfig() plugin.HandshakeConfig {
+	return plugin.HandshakeConfig{
+		ProtocolVersion:  0,
+		MagicCookieKey:   "magic-key-median-todo",
+		MagicCookieValue: "magic-value-median-todo",
```

Comment:
Will follow up on these soon here https://smartcontract-it.atlassian.net/browse/BCF-2198